slack/README.md

```diff
@@ -0,0 +1,121 @@
+# slack
+
+This library allows using Slack with Wing.
+
+## Prerequisites
+
+* [winglang](https://winglang.io)
+
+## Installation
+
+Use `npm` to install this library:
+
+```sh
+npm i @winglibs/slack
+```
+## Bring it
+
+```js
+bring slack;
+bring cloud;
+
+let myBotToken = new cloud.Secret(name: "mybot_token");
+
+let slackbot = new slack.App(botToken: myBotToken);
+
+slackbot.onEvent("app_mention", inflight (ctx, event) => {
+  let message = new slack.Message();
+  message.addSection({
+    fields: [
+      {
+        type: slack.FieldType.mrkdwn,
+        text: "*Wow this is markdown!!*\ncool beans!!!"
+      }
+    ]
+  });
+
+  ctx.channel.postMessage(message);
+});
+```
+
+### Create your Slack App
+
+1. Go to the [Slack API Dashboard](https://api.slack.com/apps) and create a new app.
+2. Select **Create from Scratch**.
+3. For the README example above, ensure you provide the following permissions:
+  - `app_mentions:read`
+  - `chat:write`
+  - `chat:write.public`
+  - `channels:read`
+4. Navigate to **Events** and subscribe to the following events:
+  - `app_mention`
+5. Navigate to **OAuth & Permissions** and install the app to your workspace
+6. Copy the **Bot User OAuth Token** to your clipboard
+7. Navigate to **Event Subscriptions** and enable events, then subscribe to bot events:
+  - `app_mention`
+
+### Running in the Wing Simulator
+
+First lets configure our Slack bot token as a secret in the simulator. 
+
+```sh
+wing secrets
+```
+
+When prompted, paste the Bot User OAuth Token you copied earlier.
+
+Next when running in the Wing Simulator, you will need to expose the endpoint of the Slackbot API, this can be done through the simulator console by selecting `Open tunnel for this endpoint` 
```

Comment:
Aren't we using websockets? Why do we need to expose?

Reply:
No I dropped the websockets impl for sim since it made the switch from local development to AWS deployments more work in the slack portal. 

(requires disabling and enabling events API as you go back and forth)

Reply:
But then the app points only to one URL, no? So you can't both keep an app in production and in development.

Maybe the "correct" thing to do is to create two apps: one for dev and one for prod?

(we need to get this right or otherwise this won't hold water for real applications)

Reply:
> But then the app points only to one URL, no?

This is the case either way, when you enable socket mode slack will stop sending requests to your URL even if its specified (it confused the hell out of me why I was not getting messages for a while)

From slack:
```
Receive app payloads via Websockets instead of Request URLs
Turning on Socket Mode will route your app’s interactions and events over a WebSockets connection instead sending these payloads to Request URLs, which are public HTTP endpoints.

This setting is intended for internal apps that are in development or need to be deployed behind a firewall. It is not intended for widely distributed apps. Please set up Request URLs for your app before submitting to the App Directory. [Learn more](https://api.slack.com/socket-mode)
```

> Maybe the "correct" thing to do is to create two apps: one for dev and one for prod?

Yes it seems like the general guidance online is to use separate apps since there is not a concept of environments for slack Apps.

Reply:
Which maintaining two apps is kinda cumbersome, since if you have to update the manifest in dev, theres no easy copy this to another app button. Its a manual copy and paste job. Which I think I understand, but still sucks